pycaw/tools/app_mixer/app_controll_and_gui.py

```diff
@@ -0,0 +1,248 @@
+        icon_path = os.path.join(dirname, 'app_mixer.ico')
+        self.iconbitmap(icon_path)
+
+        # ________________ STYLING ________________
+        s = Style()
+
+        s.configure("Active.TFrame", background='#007AD9')
+        # s.configure("Inactive.TFrame", background='#A2A2A2')
+        s.configure("Inactive.TFrame", background='#AF3118')
+
+        s.configure("Unmuted.TButton", foreground='#007AD9')
+        # s.configure("Muted.TButton", foreground='#C33C54')
+        s.configure("Muted.TButton", foreground='#AF3118')
+
+        # _________________ HEADER _________________
+        header = Frame(self)
+        title = Label(header, text="Live App Mixer",
+                      font=("Calibri", 20, "bold"))
+
+        # ________________ SEPARATE ________________
+        separate = Separator(header, orient=HORIZONTAL)
+
+        # ____________ ARRANGE ELEMENTS ____________
+        header.columnconfigure(0, weight=1)
+        title.grid(row=0, column=0, pady=10, sticky="W")
+        separate.grid(row=1, column=0, sticky="EW", pady=10)
+
+        # _______________ PACK FRAME _______________
+        header.pack(fill='x', padx=15)
+
+        # ____________ BUILD WITH PYCAW ____________
+        turbo_anonym = Label(self, text="created by TurboAnonym with pycaw",
+                             font=("Consolas", 10, "italic"))
+        turbo_anonym.pack(side=BOTTOM, fill="x", pady=6, padx=6)
+
+
+class AppRow(Frame):
+    """
+    Row for each app in mixer.
+    handles refreshing the gui if session is changed external.
+    handles user input and changing session volume/mute.
+    """
+
+    def __init__(self, RootFrameInstance, session, iid, session_manager):
+        # ______________ SAVING ARGS ______________
+        self.audio_session = session
+        # RootFrameInstance is the RootFrame instance
+        self.iid = iid
+        self.session_manager = session_manager
+        # session_manager has nothing to do with the
+        # IAudioSessionManager2 its just a name
+
+        # _________________ AUDIO _________________
+        self.simple_volume = self.audio_session.SimpleAudioVolume
+        # access to SimpleAudioVolume where you can use
+        # GetMasterVolume etc look at pycaw.pycaw
+
+        # __________________ GUID __________________
+        self.lpcguid = RootFrameInstance.lpcguid
+        # lpcguid.contents is a string like
+        # {9AC17140-FA50-4695-ACC7-1F762EC290BC}
+
+        # ______________ DISPLAY NAME ______________
+        self.app_name = self.audio_session.DisplayName
+        if not self.app_name:
+            self.app_name = self.audio_session.Process.name()
+        elif r"system32\audiosrv.dll" in self.app_name.lower():
+            # @%SystemRoot%\System32\AudioSrv.Dll,-202
+            # is displayd for system sounds
+            self.app_name = "System Sounds"
+
+        # ______________ CREATE FRAME ______________
+        Frame.__init__(self, RootFrameInstance)
```

Comment:
Just curious, why not `super().__init__(RootFrameInstance)`

Reply:
ahh i remember - it was copy pasted xD